msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
Hmm, but for most cases, clients won't know what the numbers / names of internal nodes are, and this changes for each tree in the tree sequence. So I can't see how this would be possible for a client to work out, TBH.

My need I think is the common one, which is to colour samples (tips) in such a way as to see whether they cluster. For this, you need to colour higher-level branches, without knowing beforehand what the tree shape is.

I've made the colouring more general, so that *if you do know a specific internal node*, you can choose to override the inherited colour. But I can't think that this would be used very often.